### ISSUE TITLE
Creating cache directory

### ISSUE THREAD

**Issue by gblock0** [ludovicchabant/vim-gutentags#34]
Is there a way to have Gutentags create the cache directory if it doesn't exist?

I am using the following setting, but if the folder doesn't exist then Gutentags throws an error.

`let g:gutentags_cache_dir = "~/.dotfiles/vim/vim.symlink/cache/gutentags"`

This is the error I get:
```
Error detected while processing function gutentags#setup_gutentags..<SNR>114_update_tags..gutentags#ctags#generate:
line    5:
E344: Can't find directory "/Users/gblock0/.dotfiles/vim/vim.symlink/cache/gutentags" in cdpath
Press ENTER or type command to continue
Error detected while processing function gutentags#setup_gutentags..<SNR>114_update_tags..gutentags#ctags#generate:
line    5:
E472: Command failed
Press ENTER or type command to continue
Error detected while processing function gutentags#setup_gutentags:
line   54:
E171: Missing :endif
Press ENTER or type command to continue

```

**Comment by ludovicchabant** [comment]
It's supposed to create it, but I just realized Vim doesn't take kindly to paths with `~/`, so I need to expand that path first. It's fixed locally, you'll get the fix next time I push to Github... thanks for reporting this!

**Comment by gblock0** [comment]
Thanks!

**Comment by ludovicchabant** [comment]
Fixed with da7a84ae43ff33e876026d90f3e0de9203ace200
